src/ngAria/aria.js

```diff
@@ -235,30 +251,24 @@ ngAriaModule.directive('ngShow', ['$aria', function($aria) {
           }
         },
         post: function(scope, elem, attr, ngModel) {
-          var needsTabIndex = shouldAttachAttr('tabindex', 'tabindex', elem)
-                                && !isNodeOneOf(elem, nodeBlackList);
+          var needsTabIndex = shouldAttachAttr('tabindex', 'tabindex', elem, false);
+          var ngTrueValue = $parse(attr.ngTrueValue)(scope);
 
           function ngAriaWatchModelValue() {
             return ngModel.$modelValue;
           }
 
           function getRadioReaction() {
-            if (needsTabIndex) {
-              needsTabIndex = false;
-              return function ngAriaRadioReaction(newVal) {
-                var boolVal = (attr.value == ngModel.$viewValue);
-                elem.attr('aria-checked', boolVal);
-                elem.attr('tabindex', 0 - !boolVal);
-              };
-            } else {
-              return function ngAriaRadioReaction(newVal) {
-                elem.attr('aria-checked', (attr.value == ngModel.$viewValue));
-              };
-            }
+            return function ngAriaRadioReaction(newVal) {
+              var boolVal = (attr.value == ngModel.$viewValue);
+              elem.attr('aria-checked', boolVal);
+            };
           }
-
-          function ngAriaCheckboxReaction() {
-            elem.attr('aria-checked', !ngModel.$isEmpty(ngModel.$viewValue));
+          function getCheckboxReaction() {
+            return function ngAriaCheckboxReaction(newVal) {
+              var modelValue = attr.ngTrueValue ? ngModel.$viewValue == ngTrueValue : newVal;
```

Comment:
`ngChecked` isn't restricted to `input` elements, but `ngTrue/FalseValue` are.

If they are trying to use `ngTrueValue` on a custom checkbox and it is not working (because it's not an `input`), I believe `aria-checked` is the least of their problems :)
Besides, this isn't working correctly (see https://github.com/angular/angular.js/pull/13938/files#diff-604b5030e9e44897f6ba0106fe4d28deR20) - it couldn't be, since we are assuming that `ngTrue/FalseValue` works (and compare against the parsed `ngTrueValue`), when in fact it has no effect.